eng/scripts/Copy-ApiVersion.ps1

```diff
@@ -1,77 +1,175 @@
+Updated the API version from $BaseVersion to $NewVersion.
 "@ | git commit --file=-
 
-Write-Host ''
-Write-Host '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!' -ForegroundColor 'Yellow'
-Write-Host '!!! IMPORTANT: Action Required !!!' -ForegroundColor 'Yellow'
-Write-Host '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!' -ForegroundColor 'Yellow'
-Write-Host ''
-Write-Host "You must manually update the default version and copy $FromVersion sections from any markdown files in $readmeDirectory."
-Write-Host 'When complete, commit those changes as shown below, push to the upstream repo, and submit a pull request.'
-Write-Host ''
-Write-Host '    git commit -am"Updated readme files"'
-Write-Host ''
\ No newline at end of file
+# Update readme file in the service type directory
+$jsonFiles = Get-ChildItem $newDirectory -Filter '*.json' | Select-Object -ExpandProperty Name
+$readmeFile = Get-ChildItem $readmeDirectory -Filter 'readme.md'
+
+$readmeContentBlock = @"
+### Tag: package-$newApiVersion
+
+These settings apply only when ``--tag=package-$newApiVersion`` is specified on the command line.
+
+```````yaml `$(tag) == 'package-$newApiVersion'
+input-file:
+$(
+foreach ($jsonFile in $jsonFiles) {
+  "  - $Provider/$newApiVersionStatus/$newApiVersion/$jsonFile"
+}
+)
+```````
+
+"@
+
+if ($readmeFile) {
+    $readmeContent = $readmeFile | Get-Content -Raw
+    $readmeContent -replace '(?s)(### tag: package.*)', "$readmeContentBlock`n`$1" | Set-Content $readmeFile.FullName -NoNewline
+}
+
+# Update the tag to the latest preview or stable version in the readme file
+$currentTag = $readmeContent -match '(openapi-type:.*\n+tag:\s*)(?<version>package-.*)'
+$currentTag = $Matches['version']
+$latestVersionDate = if ($currentTag -match '(\d{4})-(\d{2})-?(\d{0,2}).*') { 
+    if ($Matches[3] -eq '') { $Matches[3] = '01' }
+    [datetime]::ParseExact($Matches[1] + '-' + $Matches[2] + '-' + $Matches[3], 'yyyy-MM-dd', $null) 
+} 
+else { 
+    Write-Warning "No date found in the readme tag"
+    [datetime]::MinValue 
+}
+
+$newVersionDate = if ($newApiVersion -match '(\d{4})-(\d{2})-?(\d{0,2})') { 
+    if ($Matches[3] -eq '') { $Matches[3] = '01' }
+    [datetime]::ParseExact($Matches[1] + '-' + $Matches[2] + '-' + $Matches[3], 'yyyy-MM-dd', $null) 
+} 
+else { 
+    Write-Warning "No date found in the new version"
+    [datetime]::MinValue 
+}
+
+if ( $newVersionDate -gt $latestVersionDate) {    
+    Write-Verbose "Updating the first tag in the first yaml code block in $readmeFile"
+    if ($readmeFile) {
+        $providerReadmeContent = $readmeFile | Get-Content -Raw
+        $providerReadmeContent -replace '(openapi-type:.*\n+tag:\s*)(package-.*)', "`$1package-$newApiVersion" | Set-Content $readmeFile.FullName -NoNewline
+    }
+}
+else {
+    Write-Warning "The new version is not newer than the current version in the readme file. No changes were made."
+}
+
+git add $readmeDirectory
+git commit -am"Updated readme files"
```

Comment:
Can we change the commit to more meaningful message (probably include new version)?